### PR TITLE
WIP: Initial Kura API.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### Description of the Change

Kura library as a data storage for `Iroha2`.

### Benefits

In-memory cached disk storage.

### Possible Drawbacks 

Maintenance and support.

### CODE REVIEW

File: kura/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        account_id: String,
+        /// Quorum field (indicates required number of signatures).
+        quorum: u8,
+    }
+
+    /// A command is an intention to change the state of the network.
+    /// For example, in order to create a new role in Iroha you have to issue Create role command.
+    pub struct Command {
+        /// JSON command representation.
+        json: String,
```
Comment (takemiyamakoto) [inline kura/src/lib.rs]: This won't be a JSON string :)

Let's figure out a format for the vec<u8> of transactions.

File: kura/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+/// Representation of a consistent storage.
+struct Disk {}
+
+impl Disk {
+    fn write(&mut self, block: iroha::Block) -> Result<u64, String> {
+        Result::Err("Not implemented yet.".to_string())
+    }
+}
+
+mod iroha {
```
Comment (takemiyamakoto) [inline kura/src/lib.rs]: This won't be the iroha module. I'll push my outline code and you can then add things to the model.

File: kura/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+impl Kura {
+    /// Kura reads all transactions in all block keeping its order without any validation.
+    /// Better to use only for operations with no expectations about correctnes.
+    fn fast_init() -> Self {
+        Kura {}
+    }
+
+    /// `Kura::fast_init` with transactions and blocks validation (signatures correctness and business rules).
+    fn strict_init() -> Self {
+        Kura::fast_init()
```
Comment (takemiyamakoto) [inline kura/src/lib.rs]: Looks good. For the MVP I would implement fast_init first and then strict_init later on, once we have the MVP.

**Comment by humb1t** [comment]
Too dirty to use.

**Pull request closed without merging**